srcup/extract.py

```diff
@@ -66,10 +68,12 @@ def generate_remapping(references: list[str], file_ids: set[str]) -> dict[str, s
     }
 
 
-def process(artifact: CryticCompile) -> list[tuple[ContractSource, ContractBytecode]]:
-
-    contracts: list[tuple[ContractSource, ContractBytecode]] = []
-
+def process(artifact: CryticCompile, extra_fields: dict, use_ir: bool) -> list[tuple[ContractSource, BaseModel]]:
+    if not use_ir:
+        contracts: list[tuple[ContractSource, ContractBytecode]] = []
+    else:
+        contracts: list[tuple[ContractSource, YulIRCode]] = []
+    output_dir = os.path.join(artifact.working_dir, "out")
```

Comment:
You can improve the type-hint here and remove the if statements (`list[tuple[ContractSource, ContractBytecode | YulIRCode]]`), you could use the [typing.overload](https://docs.python.org/3/library/typing.html#typing.overload) to tie the type information to the `use_ir` argument. Making the `use_ir` optional by setting the default to `False` would also minimise changes to pre-existing code.